mapie/calibration.py

```diff
@@ -0,0 +1,549 @@
+        y_calib : NDArray of shape (n_samples,)
+            Training labels.
+        top_class_prob : NDArray of shape (n_samples,)
+            The independent values of the calibrator, it represents the
+            maximum score in the probability predictions.
+        top_class_prob_arg : NDArray of shape (n_samples,)
+            The array to determine which class the max prediction belongs to.
+        sample_weight : Optional[ArrayLike] of shape (n_samples,)
+            Sample weights. If None, then samples are equally weighted.
+
+        Returns
+        -------
+        RegressorMixin
+            Calibrated estimator.
+        """
+        calibrator_ = clone(calibrator)
+        y_calib = cast(NDArray, y_calib)
+        sample_weight = cast(NDArray, sample_weight)
+        correct_label = np.where(top_class_prob_arg.ravel() == item)[0]
+        y_calib_ = check_binary_zero_one(
+            np.equal(y_calib[correct_label], item).astype(int)
+        )
+        top_class_prob_ = top_class_prob[correct_label]
+
+        if sample_weight is not None:
+            sample_weight_ = sample_weight[correct_label]
+            (
+                sample_weight_, top_class_prob_, y_calib_
+            ) = check_null_weight(  # type: ignore
+                sample_weight_,
+                top_class_prob_,
+                y_calib_
+            )
+        else:
+            sample_weight_ = sample_weight
+        calibrator_ = fit_estimator(
+            calibrator_, top_class_prob_, y_calib_, sample_weight_
+        )
+        return calibrator_
+
+    def _get_calibrator(
+        self,
+        cv: Union[str, BaseCrossValidator],
+        calibrator: Optional[Union[str, RegressorMixin]],
+    ) -> RegressorMixin:
+        """
+        Check the input that has been provided for
+        calibrator and checks that the calibrator is a correct
+        estimator to calibrate.
+
+        Parameters
+        ----------
+        cv : Union[str, BaseCrossValidator]
+            Cross validation parameter.
+        calibrator : Union[str, RegressorMixin]
+            Calibrator as string to then be linked to one of the
+            valid methods otherwise calibrator as estimator.
+
+        Returns
+        -------
+        RegressorMixin
+            RegressorMixin to be used as calibrator.
+        Raises
+        ------
+        ValueError
+            If str is not one of the valid estimators.
+        """
+        if calibrator is None:
+            calibrator = "sigmoid"
+
+        if isinstance(calibrator, str):
+            if calibrator in self.valid_calibrators.keys():
+                calibrator = self.valid_calibrators[calibrator]
+            else:
+                raise ValueError(
+                    "Please provide a valid string from the valid calibrators."
+                )
+        check_estimator_fit_predict(calibrator)
+        return calibrator
+
+    def _get_labels(
+        self,
+        method: str,
```

Comment:
method is already in your self, so you don't need to have the method argument in your function

Reply:
done.